nss/integration-tests/integration_test.go

```diff
@@ -120,11 +152,54 @@ func TestIntegration(t *testing.T) {
 	}
 }
 
+func TestMockgpasswd(t *testing.T) {
+	grouptests.Mockgpasswd(t)
+}
+
+func testArgs(t *testing.T) (args []string) {
+	t.Helper()
+
+	args = []string{
+		os.Args[0],
+	}
+	for _, arg := range os.Args[1:] {
+		if !strings.HasPrefix(arg, "-test.gocoverdir=") {
+			continue
+		}
+		args = append(args, arg)
+	}
+	return args
+}
+
+func gpasswdMockEnv(t *testing.T, outputFilePath, groupsFilePath string) []string {
```

Comment:
isn’t that code duplicated and maybe should be moved to `grouptests` too?

Reply:
This is mainly for integration tests, so we use it in two places and only once. I don't think we should add this to the `grouptests`, since it's a very peculiar setup and should be "contained", wdyt?

Reply:
My itch is scratching here (despite having one only copy) because the env variable logic is in the tests package (in `grouptests.Mockgpasswd(t)`),so basically the logic to setup the mock and having it running is separated if I’m correct, or am I fully wrong?

Reply:
Yeah, the logic is in `withgpasswdmock.go` and protected by a build tag, but indeed it's part of the localgroups package. I'll move it to `localgroups/tests` and update the PAM setup as well.